### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by mandjevant**
Proposed changes to discord bot skeleton.

**Changes:**
- Utilpkg
- Minor restructure as proposed by https://github.com/Floor-Gang/nickreq
- Fixed imports

**Possible future changes:**
- Add authclient

### CODE REVIEW

File: internal/config.go [hunk 1]
```diff
@@ -15,52 +13,31 @@ type Config struct {
...
-}
-
-// Generate a configuration.
-func genConfig(configPath string) {
 	config := Config{
 		Token:     "",
 		Prefix:    ".mention",
 		ChannelID: "",
 		LeadDevID: "",
 		AdminID:   "",
```
Comment (dylhack) [inline internal/config.go]: This could be removed

File: internal/config.go [hunk 2]
```diff
@@ -15,52 +13,31 @@ type Config struct {
...
-	}
-
-	return config
-}
-
-// Generate a configuration.
-func genConfig(configPath string) {
 	config := Config{
 		Token:     "",
 		Prefix:    ".mention",
```
Comment (dylhack) [inline internal/config.go]: This could be made blank

**Review by dylhack — COMMENTED** [review]
Looks good.

**Comment by dylhack** [comment]
I'll be adding auth-server usage after this merge.

File: internal/config.go (hunk 1 above)
Comment (mandjevant) [reply internal/config.go]: Solved in new commit

File: internal/config.go (hunk 2 above)
Comment (mandjevant) [reply internal/config.go]: What do you mean?

Comment (dylhack) [reply internal/config.go]: The string literal ".mention" could be empty or removed which will result in an empty string.

Comment (mandjevant) [reply internal/config.go]: Ah gotya

**Review by dylhack — APPROVED** [review]

**Pull request merged**